spec/numberformat.html

```diff
@@ -1251,5 +1506,313 @@ <h1>ComputeExponentForMagnitude ( _numberFormat_, _magnitude_ )</h1>
+    </emu-clause>
+
+    <emu-clause id="sec-formatapproximately" aoid="FormatApproximately">
+      <h1>FormatApproximately ( _numberFormat_, _result_ )</h1>
+      <p>
+        The FormatApproximately abstract operation modifies _result_, which must be a List of Record values as described in PartitionNumberPattern, by adding a new Record for the approximately sign, which may depend on _numberFormat_ (which must be an object initialized as NumberFormat). The following steps are taken:
+      </p>
+      <emu-alg>
+        1. Let _i_ be an index into _result_, determined by an implementation-defined algorithm based on _numberFormat_ and _result_.
+        1. Let _approximatelySign_ be an ILND String value used to signify that a number is approximate.
+        1. If _approximatelySign_ is not empty, insert a new Record { [[Type]]: *"approximatelySign"*, [[Value]]: _approximatelySign_ } at index _i_ in _result_.
+        1. Return _result_.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-collapsenumberrange" aoid="CollapseNumberRange">
+      <h1>CollapseNumberRange ( _result_ )</h1>
+      <p>
+        The CollapseNumberRange abstract operation modifies _result_, which must be a List of Record values as described in PartitionNumberRangePattern, by removing redundant information, and returns the resulting List. The algorithm is implementation dependent, but it must not introduce ambiguity: the resulting list after modification should be unique for all pairs _x_ and _y_.
+      </p>
+      <p>
+        For example, an implementation may choose to remove the currency symbol following the range separator, to produce "$3–5" instead of "$3–$5".
+      </p>
+      <p>
+        The implementation may choose to modify symbols for grammatical correctness; for example, "0.5 miles–1 mile" may become "0.5–1 miles".
+      </p>
+      <p>
+        Returning _result_ unmodified is guaranteed to be a correct implementation of CollapseNumberRange.
+      </p>
+    </emu-clause>
+
+    <emu-clause id="sec-formatnumericrange" aoid="FormatNumericRange">
+      <h1>FormatNumericRange ( _numberFormat_, _x_, _y_ )</h1>
+
+      <p>
+        The FormatNumericRange abstract operation is called with arguments _numberFormat_ (which must be an object initialized as a NumberFormat), _x_ (which must be an Intl mathematical value), and _y_ (which must be an Intl mathematical value greater than or equal to _x_), and performs the following steps:
+      </p>
+
+      <emu-alg>
+        1. Let _parts_ be ? PartitionNumberRangePattern(_numberFormat_, _x_, _y_).
+        1. Let _result_ be the empty String.
+        1. For each element _part_ in _parts_, do
+          1. Set _result_ to the string-concatenation of _result_ and _part_.[[Value]].
+        1. Return _result_.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-formatnumericrangetoparts" aoid="FormatNumericRangeToParts">
+      <h1>FormatNumericRangeToParts ( _numberFormat_, _x_, _y_ )</h1>
```

Comment:
Can we switch to the structured headers syntax? It doesn't necessarily need to be done before this is merged, but let's do it sooner rather than later.

This applies to every new AO, I suppose.

Reply:
Acknowledged. I wrote most of this spec before structured headers were common in ECMA-402. In an effort to get this merged, I would rather make only small changes that are unlikely to require extensive further review. However, this sounds like a great editorial task to be done after merging this PR. :)